### ISSUE TITLE
how do car insertion without GUI

### ISSUE THREAD

**Issue by utc1205** [hyzhou404/HUGSIM#30]
Thanks for your hard and great work. 

I want to insert cars into scene without GUI,  Can your provide an example code?  Thanks !

**Comment by hyzhou404** [comment]
Hi, the inserted vehicle rendering code is around this line: https://github.com/hyzhou404/HUGSIM/blob/main/gaussian_renderer/__init__.py#L104

**Comment by utc1205** [comment]
Thanks for your advice, I try render 3DRealCar("2024_04_22_10_35_34 ")  in a scene,  but I found the car's axis is different from original dynamic objects.  the rendered result is like below if use its gs.pth directly. 

![Image](https://github.com/user-attachments/assets/75a9c976-c404-4ce6-beb5-b6ae9aadbd77)

then I resave xyz coordinates the gs.pth after converting its axis. 
```
(model_params, first_iter) = torch.load(vehicle_file, weights_only=False)
model_params = list(model_params)
(active_sh_degree, 
xyz, 
features_dc, 
features_rest,
feats3D,
scaling, 
rotation, 
opacity,
spatial_lr_scale,
) = model_params
R = torch.tensor([[0, 1, 0],
              [0, 0, -1],
              [-1, 0, 0]], dtype=xyz.dtype, device=xyz.device)
xyz = xyz @ R
model_params[1]=xyz-xyz.mean(dim=0)
torch.save((model_params,first_iter), vehicle_file.replace("gs.pth","gs_convert_axis.pth"))

```
the car's direction looks like normally, but the render quality is very poor.  
![Image](https://github.com/user-attachments/assets/1ee053e7-b17e-4bfd-a98a-2d7830d31076)

if the xyz coods in gs.pth will be changed, others also need to be changed? Thanks in advance.

**Comment by utc1205** [comment]
```
       model_params = list(model_params)
        
        (active_sh_degree, 
        xyz, 
        features_dc, 
        features_rest,
        feats3D,
        scaling, 
        rotation, 
        opacity,
        spatial_lr_scale,
        ) = model_params
        
        R = torch.tensor([[0, 1, 0],
                      [0, 0, -1],
                      [-1, 0, 0]], dtype=xyz.dtype, device=xyz.device)
        xyz = xyz @ R
        model_params[1]=xyz-xyz.mean(dim=0)
        
        **drot = roma.quat_wxyz_to_xyzw(rotation)
        drot = roma.unitquat_to_rotmat(drot)
        w_drot = roma.quat_xyzw_to_wxyz(roma.rotmat_to_unitquat(R @ drot))
        model_params[6] = w_drot**
        torch.save((model_params,first_iter), vehicle_file.replace("gs.pth","gs_convert_axis.pth"))
```
thanks for your reply,  I add rotation like above, the render quality has improved a little , but it look still unnatural.  

![Image](https://github.com/user-attachments/assets/dde30fd7-2d0e-4ea9-bd73-4282aa2a6780)

**Comment by hyzhou404** [comment]
I notice that you use right multiplication on xyz `xyz @ R` and left multiplication on rotation `R @ drot`. Right multiplication should be used for both xyz and rotation attributes.

**Comment by utc1205** [comment]
`def rotate_rot(gaussians, rotmat):
    new_rotation = build_rotation(gaussians._rotation)
    new_rotation = rotmat @ new_rotation
    new_quat = np.array(Rotation.from_matrix(new_rotation.detach().cpu().numpy()).as_quat())
    new_quat[:, [0,1,2,3]] = new_quat[:, [3,0,1,2]] # xyzw -> wxyz
    gaussians._rotation = torch.from_numpy(new_quat).to(gaussians._rotation.device).float()`
I use this rotate function , the rendered result is ok .

**Comment by utc1205** [comment]
![Image](https://github.com/user-attachments/assets/fb64de03-a681-4e65-8d06-d92aa96398e9)
I find there are some bright spots around rendered car model .  Is it a normal appearance?

**Comment by utc1205** [comment]
thanks for your advice, because I set same position for two dynamic object.
